bin/sdk-cli.js

```diff
@@ -2,22 +2,87 @@
 
 /** @format */
 
+/**
+ * External dependencies
+ */
 const chalk = require( 'chalk' );
 const path = require( 'path' );
 const spawnSync = require( 'child_process' ).spawnSync;
+const yargs = require( 'yargs' );
 
 const buildBlockScript = path.resolve( __dirname, 'create-scripts/block.js' );
-const [ /* node */, /* bin/sdk-cli.js */, task, ...args ] = process.argv;
+const extensionsDir = path.resolve( __dirname, '../client/gutenberg/extensions' );
+const prePackagedBlocks = [
+	'editor-notes',
+	'hello-dolly',
+];
 
-if ( ! args.length || task !== 'build-block' ) {
-	console.log( chalk.red( 'usage: npx calypso-gutenberg-sdk build-block <block>' ) );
-	process.exit(1);
-}
+const buildBlock = argv => {
+	let entryFile;
 
-spawnSync( 'node', [ buildBlockScript, ...args ], {
-	env: {
-		SKIP_FLAG_IMAGES: true,
-	},
-	shell: true,
-	stdio: 'inherit',
-} );
+	if ( argv.block ) {
+		switch ( argv.block ) {
+			case 'editor-notes':
+				entryFile = path.join( extensionsDir, 'editor-notes/index.js' );
+				break;
+			case 'hello-dolly':
+				entryFile = path.join( extensionsDir, 'hello-dolly/hello-block.js' );
+				break;
+			default:
+				console.log( chalk.red( `Unknown block "${ argv.block }" - list available blocks with "list-blocks"` ) );
+				process.exit( 1 );
+		}
+	} else if ( argv.editorJs ) {
+		entryFile = argv.editorJs;
+	} else {
+		yargs.showHelp();
+		console.log( chalk.red( 'Missing a block entryfile.' ) );
+		process.exit( 1 );
+	}
+	spawnSync( 'node', [ buildBlockScript, entryFile, ( argv.output || '' ) ], {
+		env: {
+			SKIP_FLAG_IMAGES: true,
+		},
+		shell: true,
+		stdio: 'inherit',
+	} );
+};
+
+yargs
+	.scriptName( 'calypso-gutenberg-sdk' )
+	.usage( 'Usage: $0 <command> [options]' )
+	.command( {
+		command: 'list-blocks',
+		desc: 'List pre-packaged blocks; build them with --block',
+		handler: () => console.log( prePackagedBlocks.join( '\n' ) )
+	} )
+	.command( {
+		command: 'build-block',
+		desc: 'Build a block',
+		builder: yargs => yargs.options( {
+			'block': {
+				alias: 'b',
+				description: 'Build a pre-packaged block.',
+				type: 'string',
+				choises: prePackagedBlocks,
+			},
+			'editor-js': {
+				description: 'Entry for editor side JavaScript file',
+				type: 'string',
+				coerce: path.resolve,
+			},
+			'output': {
+				alias: 'o',
+				description: 'Output directory for the built block assets.',
+				type: 'string',
+				coerce: path.resolve
+			}
```

Comment:
how about for options we have

 - `editor-js`
 - `theme-js`
 - `editor-style`
 - `theme-style`
 - `output-dir`

Reply:
We should probably go with these terms used by `WP_Block_Type()` (configured via `register_block_type()`): https://github.com/WordPress/gutenberg/blob/4f5295d1e49d28de85802bfd87862d18323c1ce0/lib/class-wp-block-type.php#L41-L71
```
editor_script	// Block type editor script handle
script		// Block type front end script handle
editor_style	// Block type editor style handle
style		// Block type front end style handle
```

@dmsnell had a good point, via Slack:
> personally although it’s consistent I think that’s a little confusing here since our _primary_ build is in many cases the `editor-script`
> yet `script` seems logically prior

Reply:
I also checked that `wp-cli scaffold block` has `--theme` and `--plugin` so I’d try to steer away from using `theme` in our CLI: 
https://wordpress.org/gutenberg/handbook/blocks/generate-blocks-with-wp-cli/#using-wp-scaffold-block

Reply:
> Yep, let's add these in follow up PRs to keep this PR more focused:

well since this PR is introducing them I don't see how this is keeping things more focussed. maybe if this PR removes the `block` param that would be enough, but I don't think introducing `block` is focused or helping us actually

Reply:
>well since this PR is introducing them I don't see how this is keeping things more focussed. 

Just I don't want to add them here yet since we can't process any input from those args yet neither.

>maybe if this PR removes the block param that would be enough, but I don't think introducing block is focused or helping us actually

👍  As per Slack convo (p1532603319000226-slack-calypso-gutenberg) going to shave out the `--block`:

> we’ll work first on getting the build script working for `--editor-js`, `--editor-css`, `--php` and other hard parts and when that’s done we can consolidate them into prettier CLI API